tasksCode/AnyType.hpp

```diff
@@ -0,0 +1,412 @@
+    // Overloaded parametrized universal constructor
+    template <typename Type>
+    AnyType(Type && value) {
+        SetValueData(value);
+        SetValueType(value);
+    };
+
+	// Copy constructor
+	AnyType(const AnyType & anyType) {
+		if (anyType.valueData == nullptr) {
+			// If anyType parameter has deactivated state
+			// deactivate state of the *this object
+			delete valueData;
+			valueData = nullptr;
+			this->valueType = anyType.valueType;
+			return;
+		}
+		SetValueData();
+		*(this->valueData) = *(anyType.valueData);
+		this->valueType = anyType.valueType;
+	};
+	
+	// Move constructor
+	AnyType(AnyType && anyType) noexcept {
+		// Deactivate state of the anyType parameter
+		this->valueData = anyType.valueData;
+		anyType.valueData = nullptr;
+		this->valueType = anyType.valueType;
+		anyType.SetValueType();
+	};
+	
+	// Copy assignment operator
+	AnyType& operator=(const AnyType & anyType) {
+		if (anyType.valueData == nullptr) {
+			// If anyType parameter has deactivated state
+			// deactivate state of the *this object
+			delete valueData;
+			valueData = nullptr;
+			this->valueType = anyType.valueType;
+			return *this;
+		}
+		SetValueData();
+		*(this->valueData) = *(anyType.valueData);
+		this->valueType = anyType.valueType;
+		return *this;
+	};
+	
+	// Move assignment operator
+	AnyType& operator=(AnyType && anyType) noexcept {
+		this->valueData = anyType.valueData;
+		anyType.valueData = nullptr;
+		this->valueType = anyType.valueType;
+		anyType.SetValueType();
+		return *this;
+	};
+	
+	// Overloaded parametrized copy assignment operator
+	template <typename Type>
+	AnyType& operator=(const Type & value) {
+		SetValueData(value);
+		SetValueType(value);
+		return *this;
+	};
+	
+	// Overloaded parametrized move assignment operator
+	template <typename Type>
+	AnyType& operator=(Type && value) {
+		SetValueData(std::forward(value));
+		SetValueType(std::forward(value));
+		return *this;
+	};
+
+	// Overloaded addition assignment operator
+	template <typename Type>
+	AnyType& operator+=(const Type & value) {
+		SetValueData(value + static_cast<Type>(*this));
+		return *this;
+	};
+	
+	// Overloaded subtraction assignment operator
+	template <typename Type>
+	AnyType& operator-=(const Type & value) {
+		SetValueData(value - static_cast<Type>(*this));
+		return *this;
+	};
+	
+	// Overloaded multiplication assignment operator
+	template <typename Type>
+	AnyType& operator*=(const Type & value) {
+		SetValueData(value * static_cast<Type>(*this));
+		return *this;
+	};
+	
+	// Overloaded division assignment operator
+	template <typename Type>
+	AnyType& operator/=(const Type & value) {
+		SetValueData(static_cast<Type>(*this) / value);
+		return *this;
+	};
+	
+	// Transform to bool, if the internal type is bool,
+	// otherwise throw bad_cast exception
+    explicit operator bool() const {
+		if (valueType != ValueType::TypeBoolean)
+			throw std::bad_cast{ };
+		return static_cast<bool>((*valueData).valueBool);
```

Comment:
This is unnecessary to use static_cast, because velue type is already bool

Reply:
![image](https://user-images.githubusercontent.com/18701555/61185780-c9428b80-a665-11e9-9a5c-c3a159d15ad8.png)
Since this is the compile time cast, during execution there will be no cast exactly. Though explicit cast improves readability.